### PR TITLE
(BEDS-205) API: added possible values for event types in notificaitons

### PR DESCRIPTION
**Pull request by remoterami**

### CODE REVIEW

File: backend/pkg/api/types/notifications.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ type NotificationDashboardsTableRow struct {
 	GroupName          string   `json:"group_name"`
 	NotificationId     uint64   `json:"notification_id"` // may be string? db schema is not defined afaik
 	EntityCount        uint64   `json:"entity_count"`
-	EventTypes         []string `json:"event_types"`
+	EventTypes         []string `json:"event_types" tstype:"'validator_offline' | 'group_offline' | 'attestation_missed' | 'proposal_success' | 'proposal_missed' | 'proposal_upcoming' | 'sync' | 'withdrawal' | 'slashed_own'| 'incoming_tx' | 'outgoing_tx' | 'transfer_erc20' | 'transfer_erc721' | 'transfer_erc1155'" faker:"oneof: validator_offline, group_offline, attestation_missed, proposal_success, proposal_missed, proposal_upcoming, sync, withdrawal, slashed_own, incoming_tx, outgoing_tx, transfer_erc20, transfer_erc721, transfer_erc1155"`
```
Comment (LuccaBitfly) [inline backend/pkg/api/types/notifications.go:40]: Slash events should be `got_slashed` and `has_slashed`. `validator_online` and `group_online` should prob also be included.

**Review by LuccaBitfly — CHANGES_REQUESTED** [review]

**Review by LuccaBitfly — APPROVED** [review]
lgtm, don't forget FE approval

File: frontend/types/api/notifications.ts [hunk 2]
```diff
@@ -43,7 +43,7 @@ export interface NotificationDashboardsTableRow {
   group_name: string;
   notification_id: number /* uint64 */; // may be string? db schema is not defined afaik
   entity_count: number /* uint64 */;
-  event_types: string[];
+  event_types: 'validator_online' | 'validator_offline' | 'group_online' | 'group_offline' | 'attestation_missed' | 'proposal_success' | 'proposal_missed' | 'proposal_upcoming' | 'sync' | 'withdrawal' | 'got_slashed' | 'has_slashed' | 'incoming_tx' | 'outgoing_tx' | 'transfer_erc20' | 'transfer_erc721' | 'transfer_erc1155';
```
Comment (MauserBitfly) [inline frontend/types/api/notifications.ts:46]: I think this should be a string.

**Review by MauserBitfly — COMMENTED** [review]
thanks for the types, but I think I found an issue

File: frontend/types/api/notifications.ts (hunk 2 above)
Comment (remoterami) [reply frontend/types/api/notifications.ts]: Not sure if I understand correctly, the point of this was to not use a `string` but a union type, right? But I see that it's not an array of event types any more, just a single value - that wasn't intended, changed it (thanks!). Was this was you meant or should I change it to something else?

Comment (MauserBitfly) [reply frontend/types/api/notifications.ts]: Yes exactly that, thanks ;-)

**Comment by MauserBitfly** [comment]
@remoterami @LuccaBitfly : Pushed a small frontend fix to fix a type error

**Review by MauserBitfly — APPROVED** [review]
LGTM

**Pull request merged**